File: C/Lista_05(Matrizes)/Matriz_Geral.c
#include <stdio.h>
#include <math.h>

int main() {

    int n;

    printf("Qual a ordem da matriz? ");
    scanf("%d", &n);

    double mat[n][n];

    for(int i=0;i<n;i++){
        for(int j=0;j<n;j++){
            printf("Elemento[%d][%d]: ", i, j);
            scanf("%lf", &mat[i][j]);
        }
    }

    double soma = 0.0;

    for(int i=0;i<n;i++){
        for(int j=0;j<n;j++){
            if(mat[i][j]>0){
                soma += mat[i][j];
            }
        }
    }

    printf("\nSOMA DOS POSITIVOS: %.1lf\n\n", soma);

    int l;

    printf("Escolha uma linha: ");
    scanf("%d", &l);

    printf("LINHA ESCOLHIDA: ");
    for(int j=0;j<n;j++){
        printf("%.1lf  ", mat[l][j]);
    }

    int c;

    printf("\n\nEscolha uma coluna: ");
    scanf("%d", &c);

    printf("COLUNA ESCOLHIDA: ");
    for(int i=0;i<n;i++){
        printf("%.1lf  ", mat[i][c]);
    }

    printf("\n\nDIAGONAL PRINCIPAL: ");
    for(int i=0;i<n;i++){
        printf("%.1lf ", mat[i][i]);
    }

    printf("\n\nMATRIZ ALTERADA:\n");

    for(int i=0;i<n;i++){
        for(int j=0;j<n;j++){
            if(mat[i][j]<0){
                mat[i][j] = pow(mat[i][j],2);
            }
        }
    }

    for(int i=0;i<n;i++){
        for(int j=0;j<n;j++){
            printf("%.1lf  ", mat[i][j]);
        }
        printf("\n");
    }

    return 0;

}

File: Python/Lista_05(Matrizes)/Matriz_Geral.py
n = int(input("Qual a ordem da matriz? "))

mat = [[0 for x in range(n)] for x in range(n)]

for i in range(0,n):
    for j in range(0,n):
        mat[i][j] = float(input(f"Elemento [{i},{j}]: "))

somaPositivos = 0

for i in range(0,n):
    for j in range(0,n):
        if mat[i][j]>0:
            somaPositivos+=mat[i][j]

print(f"\nSOMA DOS POSITIVOS: {somaPositivos:.1f}\n")

linha = int(input("Escolha uma linha: "))

print("LINHA ESCOLHIDA:", end=" ")

for j in range(0,n):
    print(f"{mat[linha][j]:.1f}", end=" ")

coluna = int(input("\n\nEscolha uma coluna: "))

print("COLUNA ESCOLHIDA:", end=" ")

for i in range(0,n):
    print(f"{mat[i][coluna]:.1f}", end=" ")

print("\n\nDIAGONAL PRINCIPAL:", end=" ")

for i in range(0,n):
    print(f"{mat[i][i]:.1f}", end=" ")

for i in range(0,n):
    for j in range(0,n):
        if mat[i][j]<0:
            mat[i][j] = mat[i][j]*mat[i][j]


print("\n\nMATRIZ ALTERADA:")

for i in range(0,n):
    for j in range(0,n):
        print(f"{mat[i][j]:.1f}", end=" ")
    print()